daffodil/parser.py

```diff
@@ -106,8 +106,8 @@ def inline_comment(self, node, children):
         return self.delegate.mk_comment(node.text, True)
 
     def condition(self, node, children):
-        'condition = _ key _ test _ value sep'
-        _, key, _, test, _, val, _ = children
+        'condition = _ key _ ((test _ value) / (existence _ boolean)) sep'
```

Comment:
I think it'd be better to keep the parsing the same but raise an exception just before `mk_cmp` when test == "?=" and val is not a bool. That would allow us to give better error messages than just a syntax error